### PR TITLE
Fixes part of #1096: Crashlytics in Domain layer.

### LINKED ISSUE #1096: Add Crashlytics Wrapper for crash reporting [GSoC'20][Milestone 1]
Addition of a Crashlytics wrapper class (and its tests) that will implement customised crash reporting for the project. It will contain methods for custom logging of non-fatal exceptions to the Firebase console. 
[Reference link](https://firebase.google.com/docs/crashlytics/customize-crash-reports?platform=android)

### PR DESCRIPTION
**Pull request by Sarthak2601**
## Explanation

Fixes part of #1096 
- Added crashlytics implementation in domain module
- Added tests for the implementation. 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: domain/src/main/java/org/oppia/domain/classify/AnswerClassificationController.kt [hunk 1]
```diff
@@ -4,6 +4,7 @@ import org.oppia.app.model.AnswerGroup
 import org.oppia.app.model.Interaction
 import org.oppia.app.model.InteractionObject
 import org.oppia.app.model.Outcome
+import org.oppia.util.logging.ExceptionLogger
```
Comment (vinitamurthi) [inline domain/src/main/java/org/oppia/domain/classify/AnswerClassificationController.kt]: This is not needed now right?

File: domain/src/test/java/org/oppia/domain/classify/AnswerClassificationControllerTest.kt [hunk 2]
```diff
@@ -485,7 +487,8 @@ class AnswerClassificationControllerTest {
   @Component(modules = [
     TestModule::class, ContinueModule::class, FractionInputModule::class, ItemSelectionInputModule::class,
     MultipleChoiceInputModule::class, NumberWithUnitsRuleModule::class, NumericInputRuleModule::class,
-    TextInputRuleModule::class, DragDropSortInputModule::class, InteractionsModule::class
+    TextInputRuleModule::class, DragDropSortInputModule::class, InteractionsModule::class,
+    TestLogReportingModule::class
```
Comment (vinitamurthi) [inline domain/src/test/java/org/oppia/domain/classify/AnswerClassificationControllerTest.kt]: Check if you still need this after removing the import from AnswerClassificationController.kt

File: domain/src/test/java/org/oppia/domain/topic/TopicControllerTest.kt [hunk 3]
```diff
@@ -1119,6 +1121,17 @@ class TopicControllerTest {
       assertThat(completedStoryList.completedStoryList[1].storyId).isEqualTo(RATIOS_STORY_ID_0)
     }
 
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testGetRevisionCard_falseInformation_returnsFailure_logsException(){
```
Comment (vinitamurthi) [inline domain/src/test/java/org/oppia/domain/topic/TopicControllerTest.kt]: Can we rename the function to be a bit clearer? maybe testGetRevisionCard_noTopicAndSubtopicId_returnsFailure_logsException ?

**Review by vinitamurthi — APPROVED** [review]
LGTM, just a couple of nits!

File: domain/src/main/java/org/oppia/domain/classify/AnswerClassificationController.kt (hunk 1 above)
Comment (Sarthak2601) [reply domain/src/main/java/org/oppia/domain/classify/AnswerClassificationController.kt]: Yep, removed.

File: domain/src/test/java/org/oppia/domain/classify/AnswerClassificationControllerTest.kt (hunk 2 above)
Comment (Sarthak2601) [reply domain/src/test/java/org/oppia/domain/classify/AnswerClassificationControllerTest.kt]: We don't, made amends :)

File: domain/src/test/java/org/oppia/domain/topic/TopicControllerTest.kt (hunk 3 above)
Comment (Sarthak2601) [reply domain/src/test/java/org/oppia/domain/topic/TopicControllerTest.kt]: Done.

**Comment by Sarthak2601** [comment]
All comments resolved, checks passed. Merging now :)

**Pull request merged**